relay/relay.go

```diff
@@ -37,3 +37,18 @@ type Hosts interface {
 	// Get returns the host:port of the best peer for the given call.
 	Get(CallFrame) string
 }
+
+// CallStats is a reporter for per-request stats.
+type CallStats interface {
+	// The call succeeded (possibly after retrying).
+	Succeeded()
+	// The RPC failed.
+	Failed(reason string)
+	// End stats collection for this RPC. Must be called exactly once.
```

Comment:
Since this is the interface that other packages provide, documentation should be written for the implementor, e.g., instead of `Must be called exactly once.`, it should be `Will be called exactly once.`

I'd also put more information here about why `End` exists (maybe specify `Success` and `Failed` are called on the call res, but `End` is called when the whole call completes, so `End` should be used for latency tracking